cache/FIFO_FH.h

```diff
@@ -334,8 +334,18 @@ bool FIFO_FHCache<TKey, TValue, THash>::construct_ratio(double FC_ratio) {
   ListNode* temp_node = m_fast_head.m_next;
   ListNode* delete_temp;
   HashMapConstAccessor temp_hashAccessor;
+
+  // Chunk management logic
+  chunks.clear();
+  assert(CHUNK_RATIO > 0);
+  int chunkSize = std::ceil(FC_size*CHUNK_RATIO);
+  bool insertNextNodeInChunk = false;
   
   while(temp_node != &m_fast_tail){
+    if (count % chunkSize == 0) {
+      insertNextNodeInChunk = true;
+    }
```

Comment:
Logic ensures that if the precise n*chunkSize node is actually deleted, that it inserts the next available (non-deleted) node